src/sandbox/cgroupFactory.go

```diff
@@ -186,6 +186,64 @@ func (cg *Cgroup) setFreezeState(state string) error {
 	}
 }
 
+// get mem usage in bytes
+func (cg *Cgroup) getMemUsage() (int64, error) {
+	usagePath := cg.Path("memory", "memory.usage_in_bytes")
+	usageRaw, err := ioutil.ReadFile(usagePath)
+	if err != nil {
+		return -1, err
+	}
+	usage, err := strconv.ParseInt(strings.TrimSpace(string(usageRaw)), 10, 64)
+	// if the mem usage cannot be read, return error
+	if err != nil {
+		return -1, err
+	}
+	return usage, nil
+}
+
+// set mem limit in bytes
+func (cg *Cgroup) getMemLimit() (int64, error) {
+	limitPath := cg.Path("memory", "memory.limit_in_bytes")
+	limitRaw, err := ioutil.ReadFile(limitPath)
+	if err != nil {
+		return -1, err
+	}
+	limit, err := strconv.ParseInt(strings.TrimSpace(string(limitRaw)), 10, 64)
+	if err != nil {
+		return -1, err
+	}
+	return limit, nil
+}
+
+// get mem limit in bytes
+func (cg *Cgroup) setMemLimit(newLimit int64) error {
+	limitPath := cg.Path("memory", "memory.limit_in_bytes")
+	// if mem limit cannot be set, return error
+	if err := ioutil.WriteFile(limitPath, []byte(fmt.Sprintf("%d", newLimit)), os.ModeAppend); err != nil {
+		return err
+	}
+
+	timeout := 5 * time.Second
+
+	start := time.Now()
+	for {
+		memLimit, err := cg.getMemLimit()
+		if err != nil {
+			return fmt.Errorf("failed to get mem limit :: %v", err)
+		}
+
+		if memLimit == newLimit {
```

Comment:
This may never happen, even if we succeed.  Take a look here: https://www.kernel.org/doc/Documentation/cgroup-v1/memory.txt

> successful write to this file does not guarantee a successful setting of
> this limit to the value written into the file. This can be due to a
> number of factors, such as rounding up to page boundaries or the total
> availability of memory on the system. The user is required to re-read
> this file after a write to guarantee the value committed by the kernel.